### DISCUSSION TITLE
Trivy 0.41.0 isn't producing vulnerability findings for a specific repository

### DISCUSSION THREAD

**Discussion by falcanta-akamai** [aquasecurity/trivy#4424] [Bugs]
### Description

Trivy is unable to get vulnerability findings for this specific repo: https://github.com/linode/manager. 

### Desired Behavior

I expected vulnerability findings to occur when I ran `trivy repo https://github.com/linode/manager`. When I downgrade to an earlier version I was able to get vulnerabilities.  

### Actual Behavior

When I ran `trivy repo https://github.com/linode/manager`, I could not get any vulnerabilities for this specific repo. 

### Reproduction Steps

```bash
1. Run the `trivy repo https://github.com/linode/manager` on my workstation. 
2. Received no results
```

### Target

Git Repository

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
$ trivy repo https://github.com/linode/manager --scanners vuln                                                                       
2023-05-17T11:09:38.410-0400	INFO	Need to update DB
2023-05-17T11:09:38.410-0400	INFO	DB Repository: ghcr.io/aquasecurity/trivy-db
2023-05-17T11:09:38.410-0400	INFO	Downloading DB...
37.00 MiB / 37.00 MiB [----------------------------------------------------------------------------------------------------------------] 100.00% 6.97 MiB p/s 5.5s
2023-05-17T11:09:46.763-0400	INFO	Vulnerability scanning is enabled
Enumerating objects: 54796, done.
Counting objects: 100% (54796/54796), done.
Compressing objects: 100% (27090/27090), done.
Total 54796 (delta 38162), reused 40253 (delta 27071), pack-reused 0
2023-05-17T11:10:22.829-0400	INFO	Number of language-specific files: 1
```

### Operating System

MacOS 13.3.1 (a) Ventura

### Version

```bash
Version: 0.41.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-05-17 12:08:01.10083692 +0000 UTC
  NextUpdate: 2023-05-17 18:08:01.10083652 +0000 UTC
  DownloadedAt: 2023-05-17 15:09:46.762621 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-03-14 00:47:02.774253754 +0000 UTC
  NextUpdate: 2023-03-17 00:47:02.774253254 +0000 UTC
  DownloadedAt: 2023-03-14 14:11:11.708775 +0000 UTC
Policy Bundle:
  Digest: sha256:d19c4c0d48ed4641862e020ff7eba7fd3ba449f66b532b09d79a6023bc65bd5b
  DownloadedAt: 2023-04-25 21:34:32.584866 +0000 UTC
```

### Checklist

- [X] Run `trivy --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Which vulnerabilities do you expect to be detected?

**Comment by DmitriyLewen** [comment]
Hello @falcantaralinode 

Looks like you need dependencies from `yarn.lock` file.
But looks like all dependencies from this file are developer dependencies.

See [this](https://aquasecurity.github.io/trivy/v0.41/docs/scanner/vulnerability/language/nodejs/#yarn) for more information.

If you want get list of dev deps => use `--skip-files package.json` flag.

Regards, Dmitriy

**Reply by falcanta-akamai** [reply]
Does trivy based on "priority" (first package.json, then yarn.lock, finally pnpm-lock.yaml)? Does it first scan package.json to see if anything is there and then scan the yarn.lock file?

**Reply by DmitriyLewen** [reply]
> Does trivy based on "priority" (first package.json, then yarn.lock, finally pnpm-lock.yaml)

Trivy doesn't have priority for files. It scans all files separately.
In this case we will parse `yarn.lock` and `pnpm-lock.yaml` (`package.json` is not parsed in `fs` mode to find depencencies). 
When Trivy finds `yarn.lock` file, we try to read `package.json` file in same directory and get non-developer dependencies. After that we merge dependencies from `yarn.lock` and `package.json` (remove dev deps from `yarn.lock` file).

**Comment by nikpivkin** [comment]
@knqyf263 I think [it's](https://github.com/aquasecurity/trivy/issues/4633) related

**Reply by knqyf263** [reply]
@falcantaralinode Are you using workspace?
